docs/code/KeyValues.md

```diff
@@ -1,149 +1,155 @@
+importance to the fact that it is a `uint`, and think these are somehow just
+numbers. This is incorrect.
+
+Most importantly, that the cardinality of the set they're enumerating is part
+of the type is critical information. 
 
 For keys, the concept of order and difference has no inherent, real meaning as
 it does for numbers, or at least, the meaning is different and highly domain
-dependent. Consider a numeric `U4` type, with values `0`, `1`, and `2`. The
-difference between `0` and `1` is `1`, and the difference between `1` and `2`
-is `1`, because they're numbers. Very well: now consider that you train a term
-transform over the input tokens `apple`, `pear`, and `orange`: this will also
-map to the keys logically represented as the numbers `0`, `1`, and `2`
-respectively. Yet for a key, is the difference between keys `0` and `1`, `1`?
-No, the difference is `0` maps to `apple` and `1` to `pear`. Also order
-doesn't mean one key is somehow "larger," it just means we saw one before
-another -- or something else, if sorting by value happened to be selected.
-
-Also: ML.NET's vectors can be sparse. Implicit entries in a sparse vector are
-assumed to have the `default` value for that type -- that is, implicit values
-for numeric types will be zero. But what would be the implicit default value
-for a key value be? Take the `apple`, `pear`, and `orange` example above -- it
-would inappropriate for the default value to be `0`, because that means the
-result is `apple`, would be appropriate. The only really appropriate "default"
-choice is that the value is unknown, that is, missing.
-
-An implication of this is that there is a distinction between the logical
-value of a key-value, and the actual physical value of the value in the
-underlying type. This will be covered more later.
-
-## As an Enumeration of a Set: `KeyValues` Metadata
-
-While keys can be used for many purposes, they are often used to enumerate
-items from some underlying set. In order to map keys back to this original
-set, many transform producing key values will also produce `KeyValues`
-metadata associated with that output column.
-
-Valid `KeyValues` metadata is a vector of length equal to the count of the
-type of the column. This can be of varying types: it is often text, but does
-not need to be. For example, a `term` applied to a column would have
-`KeyValue` metadata of item type equal to the item type of the input data.
+dependent. Consider a numeric `uint` type (specifically,
+`NumberDataViewType.UInt32`), with values `0`, `1`, and `2`. The difference
+between `0` and `1` is `1`, and the difference between `1` and `2` is `1`,
+because they're numbers. Very well: now consider that you call
+`ValueToKeyMappingEstimator.Fit` to get the transformer over the input tokens
+`apple`, `pear`, and `orange`: this will also map to the keys logically
+represented as the numbers `0`, `1`, and `2` respectively. Yet for a key, is
```

Comment:
Should ``as the numbers `0`, `1`, and `2` respectively`` be ``as the numbers `1`, `2`, and `3` respectively`` ? Since `0` is the `N/A` value?

Reply:
Yeah, that's why I said "logically," not in representation. But I made the mistake of calling them numbers. Let me call them indices or something. I have a paragraph later describing this thing...